.github/workflows/sonar-pr-report-publish.yml

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (C) 2015 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Sonar PR Report Publish
+
+on:
+  workflow_run:
+    workflows: [Sonar PR Report Request]
+    types:
+      - completed
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-sonar-kubernetes-client-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  sonar:
+    name: Sonar
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      BASE_BRANCH: main
+      PR_BRANCH: ${{ github.event.workflow_run.head_branch }}
+      PR_AUTHOR: ${{ github.event.sender.login }}
+      GITHUB_REPO: ${{ github.repository }}
+      SONAR_LOGIN_TOKEN: ${{ secrets.SONAR_LOGIN_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Shallow clones should be disabled for a better relevancy of analysis
+          fetch-depth: 0
+      - name: Setup Java 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Get PR number
+        run: |
+          PR_QUERY_RESULT=$(curl -s -H "Authorization: token $GITHUB_TOKEN" \
+            "https://api.github.com/repos/$GITHUB_REPO/pulls?head=$PR_AUTHOR:$PR_BRANCH&state=open" | jq '.[0].number')
+          if [ "$PR_QUERY_RESULT" = "null" ] || [ -z "$PR_QUERY_RESULT" ]; then
+            echo "Could not find PR number for $PR_AUTHOR:$PR_BRANCH"
+            exit 1
+          fi
+          echo "PR_NUMBER=$PR_QUERY_RESULT" >> $GITHUB_ENV
```

Comment:
I understand this part is checking that the PR exists and hasn't been merged before proceeding to publish the PR results.